PollBuddy-Server/frontend/src/pages/RegisterWithPollBuddy/RegisterWithPollBuddy.js

```diff
@@ -19,18 +77,40 @@ export default class RegisterWithPollBuddy extends Component {
           </p>
           <MDBContainer className="form-group">
             <label htmlFor="nameText">Name:</label>
-            <input placeholder="SIS Man" className="form-control textBox" id="nameText"/>
+            <input placeholder="SIS Man" className="form-control textBox" id="nameText" 
+              onChange={(evt) => { this.setState({username: evt.target.value}); }}/>
+            {!this.state.userValid && 
+              <div className="error">
+                <p>Username must be between 3 and 32 characters</p>
+                <p>Valid characters: (A-Z), (0-9), (-,_,.)</p>
+              </div>
+            }
             <label htmlFor="emailText">Email:</label>
-            <input placeholder="mans@rpi.edu" className="form-control textBox" id="emailText"/>
+            <input placeholder="mans@rpi.edu" className="form-control textBox" id="emailText"
+              onChange={(evt) => { this.setState({email: evt.target.value}); }}/>
+            {!this.state.emailValid && 
+              <div className="error">
+                <p>Invalid email format!</p>
+              </div>
+            }
+            {this.state.emailExists && <div className="error">A user with this email already exists!</div>}
             <label htmlFor="passwordText">Password:</label>
-            <input placeholder="●●●●●●●●●●●●" className="form-control textBox" id="passwordText"/>
+            <input type="password" placeholder="●●●●●●●●●●●●" className="form-control textBox" id="passwordText"
+              onChange= {(evt) => { this.setState({password: evt.target.value}); }}/>
+            {!this.state.passValid && 
+              <div className="error">
+                <p>Invalid password. Must contain:</p>
+                <p>6 or more characters</p>
+                <p>At least 1 uppercase letter</p>
+                <p>At least 1 number</p>
```

Comment:
Could be good as a bulleted list perhaps, not sure. Problem for later when the requirements are figured out though.